File: java/src/main/java/algorithm/utils/Validation.java
package algorithm.utils;

import algorithm.utils.errors.parameters.NegativeParameterError;
import algorithm.utils.errors.parameters.ZeroParameterError;

public final class Validation {
    private Validation() {}

    public static void validatePositive(double value, String name) throws Exception {
        if (value < 0.0) {
            throw new NegativeParameterError(name, value);
        } else if (value == 0.0) {
            throw new ZeroParameterError(name, value);
        }
    }
}

File: rust/src/utils/validation.rs
use std::error::Error;

use super::errors::parameters::{NegativeParameterError, ZeroParameterError};

pub(crate) fn validate_positive(value: f64, name: &str) -> Result<(), Box<dyn Error>> {
    let name = name.to_string();
    match value {
        | v if v < 0.0 => Err(Box::new(NegativeParameterError { name, value })),
        | 0.0 => Err(Box::new(ZeroParameterError { name, value })),
        | _ => Ok(()),
    }
}
